ext/RastersMakie.jl

```diff
@@ -0,0 +1,276 @@
+- `colorrange = Makie.automatic`: The colormap for the heatmap.  This can be set to a vector of `(low, high)` if plotting a 3D raster or RasterStack.
+- `nancolor = :transparent`: The color which `NaN` values should take.  Default to transparent.
+"""
+function Rasters.rplot(position::GridPosition, raster::AbstractRaster{T,2,<:Tuple{D1,D2}};
+    plottype = Makie.Heatmap,
+    axistype = Makie.Axis,
+    X=X, Y=Y,
+    draw_colorbar = true,
+    colorbar_position = Makie.Right(),
+    colorbar_padding = Makie.automatic,
+    title = Makie.automatic,
+    xlabel = Makie.automatic,
+    ylabel = Makie.automatic,
+    colorbarlabel = Makie.automatic,
+    nancolor = :transparent,
+    colormap = nothing,
+    colorrange = Makie.automatic,
+    kw_attributes...
+    ) where {T,D1<:Rasters.SpatialDim,D2<:Rasters.SpatialDim}
+
+    # handle extra kwargs
+    attributes = merge(
+        Attributes(kw_attributes),
+        Attributes(;
+            Colorbar = (; label = colorbarlabel,),
+            nancolor,
+        )
+    )
+
+    isnothing(colormap) && (colormap = get(attributes, :colormap, :viridis))
+
+    # x and y labels
+    xlabel_str, ylabel_str = Rasters.label(DimensionalData.dims(raster))
+    xlabel isa Makie.Automatic && (xlabel = xlabel_str)
+    ylabel isa Makie.Automatic && (ylabel = ylabel_str)
+
+    # colorbar label
+    colorbarlabel isa Makie.Automatic && (colorbarlabel = ""#=string(DimensionalData.name(raster))=#)
+    
+    # title
+    if title == Makie.automatic
+        _rdt = DimensionalData.refdims_title(raster; issingle = true)
+        title = (_rdt === "" ? Rasters._maybename(raster) : Rasters._maybename(raster) * "\n" * _rdt)
+    end
+    
+    local axis, plot
+    # actually plot
+    with_theme(attributes) do
+        axis = axistype(position; 
+            title, xlabel, ylabel 
+        )
+        # plot to the axis with the specified plot type
+        plot = plot!(plottype, axis, raster; colormap, colorrange, nancolor)
+
+        if draw_colorbar
+
+            colorbar = Colorbar(
+                position.layout[position.span.rows, position.span.cols, colorbar_position],
+                plot;
+                label = colorbarlabel,
+            )
+
+            colorbar_padding = if colorbar_padding isa Makie.Automatic
+                    if colorbar_position in (Makie.Top(), Makie.Bottom())
+                        to_value(colorbar.layoutobservables.computedbbox[].widths[2])
+                    else
+                        to_value(colorbar.layoutobservables.computedbbox[].widths[1])
+                    end
+                else
+                    to_value(colorbar_padding)
+                end
+
+            colorbar.alignmode[] = if colorbar_position == Makie.Top()
+                Makie.Mixed(Makie.GridLayoutBase.RectSides{Union{Nothing, Float32, Makie.GridLayoutBase.Protrusion}}(nothing, nothing, colorbar_padding, 0f0))
```

Comment:
Seems like a lot of duplication of this same line, you could define

`T = Makie.GridLayoutBase.RectSides{Union{Nothing, Float32, Makie.GridLayoutBase.Protrusion}}`